workspace/notebook/py_sb_harmonised_nsip_meeting.json

```diff
@@ -0,0 +1,386 @@
+					"    aligned_columns = []\n",
+					"    for col_name in source_df.columns:\n",
+					"        if col_name in target_schema:\n",
+					"            # Column exists in target - cast to target datatype\n",
+					"            aligned_columns.append(\n",
+					"                col(col_name).cast(target_schema[col_name]).alias(col_name)\n",
+					"            )\n",
+					"        else:\n",
+					"            # Column NOT in target - keep original datatype\n",
+					"            aligned_columns.append(col(col_name))\n",
+					"    \n",
+					"    return source_df.select(aligned_columns)"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "code",
+				"metadata": {
+					"collapsed": false
+				},
+				"source": [
+					"#Read target table for alignment\n",
+					"target_df = spark.table(spark_table_final)\n",
+					"\n",
+					"#Flatten nested meeting columns and prepare bronze data\n",
+					"service_bus_exploded = service_bus_exploded.select(\n",
+					"    \"NSIPProjectInfoInternalID\",\n",
+					"    \"caseId\",\n",
+					"    \"caseReference\",\n",
+					"    col(\"meeting.meetingId\").alias(\"meetingId\"),\n",
+					"    col(\"meeting.meetingAgenda\").alias(\"meetingAgenda\"),\n",
+					"    col(\"meeting.planningInspectorateRole\").alias(\"planningInspectorateRole\"),\n",
+					"    col(\"meeting.meetingDate\").alias(\"meetingDate\"),\n",
+					"    col(\"meeting.meetingType\").alias(\"meetingType\"),\n",
+					"    \"Migrated\",\n",
+					"    \"ODTSourceSystem\",\n",
+					"    \"SourceSystemID\",\n",
+					"    \"IngestionDate\"\n",
+					")\n",
+					"\n",
+					"service_bus_exploded = align_source_to_target_schema(service_bus_exploded,target_df)\n",
+					"\n",
+					"\n",
+					"# Create business key and row hash\n",
+					"# Use ALL columns that uniquely identify a record\n",
+					"business_key_cols = [\"NSIPProjectInfoInternalID\", \"caseId\", \"meetingId\"]\n",
+					"change_tracking_cols = [ \"meetingAgenda\",\n",
+					"    \"planningInspectorateRole\", \"meetingType\", \"meetingDate\"]\n",
+					"\n",
+					"source_df = service_bus_exploded \\\n",
+					"    .withColumn(\"business_key\", sha2(concat_ws(\"~\", *business_key_cols), 256)) \\\n",
+					"    .withColumn(\"row_hash\", sha2(concat_ws(\"~\", *(business_key_cols + change_tracking_cols)), 256))\n",
+					"\n",
+					"# Deduplicate source - keep latest version if duplicates exist in same load\n",
+					"window_dedup = Window.partitionBy(\"business_key\").orderBy(col(\"IngestionDate\").desc())\n",
+					"source_df_deduped = source_df \\\n",
+					"    .withColumn(\"rn\", row_number().over(window_dedup)) \\\n",
+					"    .filter(\"rn = 1\") \\\n",
+					"    .drop(\"rn\")"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "code",
+				"source": [
+					"\n",
+					"# Check if target table exists\n",
+					"try:\n",
+					"    existing_target = spark.read.format(\"delta\").table(spark_table_final)\n",
```

Comment:
If the table does not exist existing_target is never set and you'll get NameError.
should be if not exist > initial load
else proceed with SCD2 logic